### ISSUE TITLE
Byline / sub-heading pattern

### ISSUE THREAD

**Issue by nottrobin** [ubuntudesign/vanilla-design#47]
When considering patterns needed for the blog theme, we noticed that a few different blog platforms make use of subheadings - e.g. for displaying author name / published date. For example, from insights.ubuntu.com:

![subheading-insights](https://user-images.githubusercontent.com/519935/29143784-4ae346ea-7d4e-11e7-9d62-e9200d8819c4.png)
![subheading-insights-2](https://user-images.githubusercontent.com/519935/29143786-4dc6d084-7d4e-11e7-8012-83379fd97510.png)

If we were to support this pattern in our blogs, it should certainly be its own pattern, distinct from any of the existing headings, as it's not a heading, and should be deliberately designed.

However, maybe the decision on whether to include this should wait until we decide if it's something we need for the blog theme, as I'm not sure it should be used by any of our existing types of website. Sub-headings like this would, I would imagine, only ever apply to the page title, not any other headings. And for all our existing pages, the page title tends to be quite deliberately crafted. I'm not sure whether we need the ability to add a sub-heading.

Once we have decided if this pattern should be included in the framework, from a design perspective, there will then be a bunch of development considerations for which I have ideas, but we should wait until that decision is made and then discuss those over on the vanilla-framework repository.

**Comment by matthewpaulthomas** [comment]
Subheadings and bylines are not the same thing. This is demonstrated by many publications that consistently use both at once. For example, [UX Mag](http://uxmag.com/articles/tech-companies-create-addicts):
`<h1>Tech Companies: If You Create Addicts, You Need to Help Them</h1>`
…`<div class="panel-pane pane-entity-field pane-node-field-long-article-teaser">`…`How could tech companies help people become less addicted with their products? And should they?`…`</div>`
`<div class="panel-pane pane-custom pane-1"><p>Article No :1685 | June  16, 2017 | by Nir Eyal</p></div>`

[Vanity Fair](https://www.vanityfair.com/news/2017/08/one-of-ubers-biggest-investors-just-sued-travis-kalanick-for-fraud):
`<h1 class="hed">One of Uber’s Biggest Investors Just Sued Travis Kalanick for Fraud</h1>`
`<div class="dek">Is this the end of Kalanick’s comeback?</div>`
`<div class="details-wrapper"><div class="contrib-wrapper"><div class="byline-container">…`

And [Esquire](http://www.esquire.com/entertainment/music/a56737/the-emoji-movie-review/):
`<h1 class="content-header--title standard-article--title anim-in-view in-view" itemprop="headline"><em>The Emoji Movie</em> Is Perfect If You Enjoy Seeing the Disappointed Faces of Children</h1>`
`<h2 class="content-header--dek standard-article--dek anim-in-view in-view" itemprop="alternativeHeadline"><p>Time to take a break from anyone younger than 20.`…`</p></h2>`
`<div class="byline ">…`

So, I suggest that we have a separate pattern for byline. That pattern would include layout for an optional avatar, as well as allowing author name(s) and/or date(s).

**Comment by yaili** [comment]
Some notes:

While in these examples this specific style (about same size as paragraph text, lighter grey text colour) is being used for bylines, it is also used on other sites for other bits of text

For example, on this page https://www.ubuntu.com/support/plans-and-pricing on "Per node" pricing, the "per year" and "a day per node" texts were originally lighter grey. I think I've also see instances where only the date is a lighter grey, similar to the news row on this page https://www.ubuntu.com/cloud

With this in mind, wouldn't it be better/more flexible to have a pattern/utility that changes the text colour, regardless of content type? 🤔

**Comment by nottrobin** [comment]
> With this in mind, wouldn't it be better/more flexible to have a pattern/utility that changes the text colour, regardless of content type? :thinking:

It would definitely be more flexible, and I believe that is exactly why is it *worse*.

As I say, we should decide if we need subheadings or bylines (and if @matthewpaulthomas says they're different things, I'm happy to accept that). If we do need them, then they should be patterns.

We shouldn't just be allowing people to decide they want random paragraphs to be grey without having a semantic reason for doing so. To me, this goes against the whole point of defining patterns.

**Comment by anthonydillon** [comment]
We should develop patterns when ever we can over utilities. This will
result in a much more conhesive framework. Easier to document and use
without full knowledge of all aspects of the framework.

I would like to have two new patterns (subheading, byline) with details
usage documentation.

**Comment by nottrobin** [comment]
> I would like to have two new patterns (subheading, byline) with details usage documentation.

I agree they should be separate patterns if they exist. But do we know they need to exist yet? We shouldn't be pre-emptively creating patterns we don't need.

**Comment by anthonydillon** [comment]
Yes that's true, all new patterns should be proposed via the working group
with a number of examples of usage.

In this case it is slightly different as it's the incarnation of a suite of
patterns for a blogging platform.

I think we should UX/Design a blog layout with a number of situations and
define the patterns from the new design.

**Comment by matthewpaulthomas** [comment]
There are plenty of examples of subheadings/deks on Vanilla brochure sites, using a variety of markup, which may be a sign that there should be a pattern.
- [Ubuntu Developer](https://developer.ubuntu.com/): “What will you do with Ubuntu?”
- [Ubuntu Cloud](https://www.ubuntu.com/cloud): “The bare metal physical cloud”
- [Ubuntu Cloud training](https://www.ubuntu.com/cloud/training): “Online training by Hastexo”
- [Ubuntu Server](https://www.ubuntu.com/server): “Agile, secure, deploy-anywhere technology for fast-moving companies”
- [Ubuntu Desktop for enterprise](https://www.ubuntu.com/desktop/enterprise): “Stay up-to-date with regular updates and upgrades”
- [Juju](https://jujucharms.com/): “Operate big software on any cloud” (after a main heading that is present but hidden)

Subheadings are unfortunately one of the details where WhatWG HTML (which browser vendors pay attention to) differs from W3C HTML (which many authors still pay attention to). [WhatWG recommends using the next-level heading inside an `hgroup` element](https://html.spec.whatwg.org/multipage/sections.html#the-hgroup-element):
```html
<hgroup>
 <h1>Dr. Strangelove</h1>
 <h2>Or: How I Learned to Stop Worrying and Love the Bomb</h2>
</hgroup>
```
[W3C instead recommends a `p` or `span` inside a `header` element](https://www.w3.org/TR/html5/common-idioms.html#sub-head):
```html
<header>
   <p>Magazine of the Decade</p>
   <h1>THE MONTH</h1>
   <p>The Best of UK and Foreign Media</p>
</header>
```
The difference is apparently theoretical, since [browsers currently don’t generate outlines either way](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Using_HTML_sections_and_outlines). But for the same reason, nothing goes wrong when you use either; and it would seem to be easy to write CSS selectors that catch both.

**Comment by kwm14** [comment]
### Vanilla working group

- Team agreed that 'byline' will be proposed as a new pattern
- To be designed as a new component
